### ISSUE TITLE
Need a rule around not passing a primitive to Objects.isNull()

### ISSUE THREAD

**Issue by nmolchan** [pmd/pmd#5304]
**Proposed Rule Name:**

primitiveToObjects

**Proposed Category:** One of [Best Practices | Code Style | Design | Documentation | Error Prone | Multithreading | Performance | Security]

Security | Error Prone

**Description:**

Passing a primitive to Objects.isNull() is an error, because it always returns false

**Code Sample:** This should include code, that should be flagged by the rule. If possible, the "correct" code
according to this new rule should also be demonstrated.

```java
int m_retries;
...
      return Objects
         .isNull(m_retries) ? ConstantUtils.DEFAULT_MAX_RETRIES : m_retries;

```

**Possible Properties:**

*   Should this rule be customizable via properties?

No.
